shared_model/interfaces/queries/impl/account_detail_pagination_meta.cpp

```diff
@@ -8,8 +8,11 @@
 using namespace shared_model::interface;
 
 bool AccountDetailPaginationMeta::operator==(const ModelType &rhs) const {
+  auto const &frid = firstRecordId();
+  auto const &rfrid = rhs.firstRecordId();
   return pageSize() == rhs.pageSize()
-      and firstRecordId() == rhs.firstRecordId();
+      and static_cast<bool>(frid) == static_cast<bool>(rfrid)
+      and (not frid or frid->get() == rfrid->get());
```

Comment:
why not make it reusable through a common header function like `optionalReferenceEqual`?